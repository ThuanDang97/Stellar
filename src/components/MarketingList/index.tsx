// Library
import React, { memo } from 'react'

// Themes
import { useTheme } from 'styled-components'

// Components
import LabelBox from '@components/LabelBox'
import Button from '@components/Button'
import { BoxShadowStyled } from '@components/styled-components'
import { ViewStyled } from '@components/styled-components/ViewStyled'

// Constants
import { LABEL_BOX } from '@constants/index'

// Props type
import {
  MarketingProps,
  MarkettingItemProps,
} from '@self-types/components/Marketing.props'

// Utils
import { selectItem } from '@utils/index'
import { pxToRem } from '@utils/theme'

const MarketingList = ({
  handleContinueButton,
  selectedItem,
  setSelectedItem,
}: MarketingProps) => {
  const theme = useTheme()
  const isDisabled = selectedItem?.length === 0

  return (
    <BoxShadowStyled
      maxWidth={380}
      padding={theme.metrics.dimensions.xmd}
      display="flex"
      flexDirection="column"
      alignItems="initial"
      flexWrap="wrap"
    >
      <ViewStyled
        display="flex"
        flexWrap="inherit"
        pBottom={theme.metrics.dimensions.lg}
        justifyContent="center"
      >
        {LABEL_BOX.map((name: MarkettingItemProps) => {
          const handleMarketingItem = () =>
            selectItem(setSelectedItem, selectedItem, name.id)

          return (
            <ViewStyled
              padding={`${pxToRem(theme.metrics.dimensions.xs)}`}
              key={`labelBox-${name.id}`}
            >
              <LabelBox
                title={name.title}
                isActive={selectedItem.includes(name.id)}
                handleOnClick={handleMarketingItem}
                padding={`${pxToRem(theme.metrics.dimensions.sm)} ${pxToRem(
                  theme.metrics.dimensions.md,
                )}`}
              />
            </ViewStyled>
          )
        })}
      </ViewStyled>
      <Button
        {...(!isDisabled && { cursor: 'pointer' })}
        onClick={handleContinueButton}
        title="Continue"
        variant="colorDefault"
        disabled={isDisabled}
      />
    </BoxShadowStyled>
  )
}

export default memo(MarketingList)
