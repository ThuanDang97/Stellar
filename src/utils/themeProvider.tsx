import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
