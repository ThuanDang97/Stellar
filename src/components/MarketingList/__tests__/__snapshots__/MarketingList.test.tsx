// Library
import { cleanup } from '@testing-library/react'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Components
import MarketingList from '@components/MarketingList'

describe('Marketing render', () => {
  afterEach(cleanup)

  const component = renderWithTheme(
    <MarketingList setSelectedItem={jest.fn()} selectedItem={[]} />,
  )

  it('should render marketing component', () => {
    const { asFragment } = component

    expect(asFragment()).toMatchSnapshot()
  })
})
