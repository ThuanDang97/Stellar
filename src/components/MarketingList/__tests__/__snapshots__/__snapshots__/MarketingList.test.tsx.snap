// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marketing render should render marketing component 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 kRzxHL dbZetz"
    display="flex"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 fEXkeO"
      display="flex"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            Podcast
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            Referral
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            Clubhouse
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            Pinterest
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            YouTube
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            Facebook
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            Online Ad
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            Wed Search
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            Instagram
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            TikTok
          </span>
        </div>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 gWlwSj"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 hBwktS"
          cursor="pointer"
          data-testid="label-box"
        >
          <span
            class="TextStyled-sc-9xia41-0 dItzST"
            color="#242A30"
            font-size="14"
          >
            Other
          </span>
        </div>
      </div>
    </div>
    <button
      class="ButtonStyled-sc-1ys9v52-0 xbTIZ"
      data-testid="button"
      disabled=""
    >
      Continue
    </button>
  </div>
</DocumentFragment>
`;
